### PR TITLE
fix: [M3-7624] - AGLB Configuration - Save button disabled when trying to remove a route

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝
- The AGLB Configuration "Save Changes" button remained disabled even if they tried to remove a route from a configuration

## Preview 📷

| Before  | After   |
| ------- | ------- |
| "Save Changes" never becomes enabled after removing a route | The "Save Changes" button becomes enabled when a change is made | 
|  <video src="https://github.com/linode/manager/assets/115251059/374a5bb3-a4b0-4ec2-a9a5-3128747fd96d" /> | <video src="https://github.com/linode/manager/assets/115251059/870728cf-ad5b-41cb-ae1a-7c92da610a67" /> |

### Prerequisites
- Use the dev environment
- Sign into the `dev-test-aglb` account
- Test using any AGLB on the test account

### Reproduction steps
- Go to the configurations tab on an AGLB
- Remove a route from a configuration using the action menu
- Observe that when you remove the route, you are unable to save changes

### Verification steps 
- Go to the configurations tab on an AGLB
- Remove a route from a configuration using the action menu
- Verify the "Save Changes" button becomes enabled
- Verify clicking Save Changes fires off a PUT, with the correct routes in the payload

> [!Warning]
> Due to an API bug, removing a route on a configuration does not currently work. We can still review this PR by just verifying the "Save Changes" button becomes active when it should

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationForm.tsx [hunk 1]
```diff
@@ -128,8 +128,9 @@ export const ConfigurationForm = (props: CreateProps | EditProps) => {
     if (!formik.values.route_ids) {
       return;
     }
-    formik.values.route_ids.splice(index, 1);
-    formik.setFieldValue('route_ids', formik.values.route_ids);
```
Comment (bnussman-akamai) [inline packages/manager/src/features/LoadBalancers/LoadBalancerDetail/Configurations/ConfigurationForm.tsx:132]: Updating the state this way is unreliable. Copying the array ensures the proper rerender happens

**Review by mjac0bs — APPROVED** [review]
Looks good with the button, as far as the behavior we can test. ++ to the added 🧪 coverage.

| Request | Response |
| --- | --- |
| Screenshot 2024-01-09 at 2 56 28 PM | Screenshot 2024-01-09 at 2 56 21 PM |

**Comment by mjac0bs** [comment]
Just noticed the test failure on `loadbalancer-configurations.spec`. Is that legit?

**Comment by bnussman-akamai** [comment]
Oops! I made a silly mistake @mjac0bs. Fixed in bf0503ba80

**Pull request merged**